### PR TITLE
Upgrade react-native to 0.59.0

### PR DESCRIPTION
**Pull request by tbergq**
Working on ios
Working on android

Upgrade react-native-maps to v0.24.0-rc1

### CODE REVIEW

File: packages/shared/package.json [hunk 1]
```diff
@@ -5,26 +5,26 @@
...
     "@kiwicom/mobile-navigation": "^0",
     "@kiwicom/mobile-orbit": "^0",
-    "@kiwicom/mobile-config": "^0",
     "@kiwicom/rnmodules": "^0",
     "@ptomasroos/react-native-multi-slider": "^1.0.0",
     "prop-types": "^15.6.2",
     "querystring": "^0.2.0",
     "react-native-image-progress": "^1.1.1",
-    "react-native-maps": "0.21.0",
+    "react-native-maps": "github:react-native-community/react-native-maps#6c225af612d4681a00d4849b60774d7c6426835e",
```
Comment (tbergq) [inline packages/shared/package.json]: This was the commit has marking the `0.0.24-rc1`

Comment (tbergq) [reply packages/shared/package.json]: And also, 0.23.0 does not work with rn 0.59

**Comment by tbergq** [comment]
also note, I am merging into a feature branch, I might want to do some more native upgrades before we merge this into master.

File: app/hotels/src/map/allHotels/MapView.js [hunk 2]
```diff
@@ -175,7 +165,15 @@ export class Map extends React.Component<Props, State> {
       return;
     }
 
-    this.map.animateToCoordinate(coordinate, duration);
+    this.map.animateCamera(
```
Comment (RobinCsl) [inline app/hotels/src/map/allHotels/MapView.js]: changes in API, I see.

Comment (tbergq) [reply app/hotels/src/map/allHotels/MapView.js]: Yes, animateToCoordinate has been deprecated

File: packages/localization/src/Alert.js [hunk 3]
```diff
@@ -37,8 +37,6 @@ type AlertOptions = {
 
 type AlertButtonStyle = 'default' | 'cancel' | 'destructive';
 
-type AlertType = 'default' | 'plain-text' | 'secure-text' | 'login-password';
```
Comment (RobinCsl) [inline packages/localization/src/Alert.js]: Why are we removing this?

File: packages/shared/package.json [hunk 4]
```diff
@@ -5,26 +5,26 @@
...
     "react-native-modal": "^7.0.0",
     "react-native-read-more-text": "^1.1.0"
   },
   "peerDependencies": {
+    "@kiwicom/universal-components": "^0",
```
Comment (RobinCsl) [inline packages/shared/package.json]: If we're using TextInput from universal, there were some changes in prop names in v0.0.16.

Comment (RobinCsl) [reply packages/shared/package.json]: We definitely need to get better at maintaining some kind of changelog with breaking changes, etc. I will create an issue for it.

File: packages/localization/src/Alert.js (hunk 3 above)
Comment (tbergq) [reply packages/localization/src/Alert.js]: Using alert type was deprecated, and has been removed in v 0.59

File: packages/shared/package.json (hunk 4 above)
Comment (tbergq) [reply packages/shared/package.json]: Ok, thanks for the heads up, I am still on 0.0.14, and we don't have any textInput I think

Comment (RobinCsl) [reply packages/shared/package.json]: https://github.com/kiwicom/margarita/issues/595

File: packages/localization/src/Alert.js (hunk 3 above)
Comment (tbergq) [reply packages/localization/src/Alert.js]: https://github.com/facebook/react-native/commit/e2bd7db

**Pull request merged**